src/components/with-search/index.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { memo, FC, ComponentType } from 'react';
+import { compose, bindActionCreators, Dispatch } from 'redux';
+import { connect } from 'react-redux';
+
+import * as actions from './redux/actions';
+
+import type { SearchResult } from '../../types';
+
+export interface Props {
+  searchResults: SearchResult;
+  searchActions: typeof actions;
+}
+
+const withSearch = (Component: ComponentType<any>) => {
```

Comment:
Tror det kan være bedre å endre hele denne komponenten til withDatasets og ha komponenten dedikert for datasett.

Reply:
Hvis du gjør det så endre alle tilfeller av "search" til "dataset". F.eks. "DATASETS_REQUESTED", datasetsRequested, datasetsSucceeded osv.